SocketServer.py

```diff
@@ -0,0 +1,105 @@
+import socket
+import termios, tty, sys
+import threading
+import pickle
+from writeCSV import WriteCSV
+
+# A thread class from https://www.tutorialspoint.com/python/python_multithreading.htm
+# This class represents a background thread used by the server to store collected data into a .csv file
+class CSVBackgroundThread (threading.Thread):
+   def __init__(self, threadID, name, counter, writer):
+        threading.Thread.__init__(self)
+        self.threadID = threadID
+        self.name = name
+        self.counter = counter
+        self.writer = writer
+   def run(self):
+        # Printing and writing to CSV file
+        while True:
+            data = conn.recv(4096)
+            listOfValues = pickle.loads(data)
+            print('ls:%0.3f rs:%0.3f lu:%0.3f ru:%0.3f lm:%0.3f rm:%0.3f' % (listOfValues[0], listOfValues[1], listOfValues[2], listOfValues[3], listOfValues[4], listOfValues[5]))
+            self.writer.writeData(listOfValues)
+
+# ==== CSV FUNCTIONS ==== #
+# ==============================================
+
+def startNewThread(name, writer):
+    # Create a new daemon thread just for taking in sensory-motor values
+    thread1 = CSVBackgroundThread(1, "Thread-1", 1, writer)
+    thread1.daemon = True
+    thread1.start()
+
+# ==============================================
+
+def openCSVFile():
+    writer = WriteCSV('output.csv')
+    writer.openFile()
+    return writer
+
+# ==============================================
+
+# ==== ROBOT COMMAND FUNCTIONS ==== #
+# ==============================================
+def getch():
+    fd = sys.stdin.fileno()
+    old_settings = termios.tcgetattr(fd)
+    tty.setcbreak(fd)
+    ch = sys.stdin.read(1)
+    termios.tcsetattr(fd, termios.TCSADRAIN, old_settings)
+
+    return ch
+
+
+# ==============================================
+
+# Code is based on https://stackoverflow.com/questions/41294848/python-sockets-how-to-connect-between-two-computers-on-the-same-wifi
+def Main():
+
+    #This IP address allows it to broadcast it to "all computers" on the network.
+    host = '0.0.0.0'
+    port = 5000
+
+    mySocket = socket.socket()
+    mySocket.bind((host, port))
+
+    mySocket.listen(1)
+    global conn
+    conn, addr = mySocket.accept()
+
+    # Setting up the .csv file
+    print("Opening output.csv")
+    writer = openCSVFile()
+    writer.writeHeader()
```

Comment:
Need to check if the file already exists or not. If it does already exist with previous values, new values (as well as a new header row) are appended to the file

Reply:
Alright did that in commit 83def8e "Added functionality to clear file before writing to .csv file... "
You can change that if you want, its in the writeCSV file (which I was reluctant to change, but oh well)